src/minigames/final/typingMinigame.ts

```diff
@@ -0,0 +1,191 @@
+      col.style.display = "flex";
+      col.style.flexDirection = "column";
+      col.style.alignItems = "center";
+      col.style.justifyContent = "center";
+      col.style.textAlign = "center";
+      col.style.gap = "4px";
+    }
+
+    const canvasBB = this.app.canvas.getBoundingClientRect();
+    const screenWidth = this.app.screen.width;
+    const screenHeight = this.app.screen.height;
+    const postWidth = screenWidth * 0.22;
+    this.postText.style.padding = "0";
+    this.postText.style.position = "absolute";
+    this.postText.style.height = `${screenHeight * 0.27}px`;
+    this.postText.style.width = `${postWidth}px`;
+    this.postText.style.left =
+      canvasBB.left +
+      screenWidth / 2 -
+      postWidth / 2 +
+      screenWidth * 0.027 +
+      "px";
+    this.postText.style.top = screenHeight * 0.63 + "px";
+    this.postText.style.position = "absolute";
+    this.postText.style.fontFamily = `'Poppins'`;
+    this.postText.style.fontSize = "var(--font-size-x-sm)";
+    this.postText.style.color = "#58676f";
+    this.postText.style.textAlign = "center";
+    this.postText.style.margin = "0";
+    this.leftCol.style.left = `${canvasBB.left}px`;
+    this.rightCol.style.right = `${canvasBB.right - screenWidth}px`;
+
+    document.body.append(this.leftCol, this.rightCol, this.postText);
+  }
+
+  override detach() {
+    super.detach();
+
+    this.leftCol.remove();
+    this.rightCol.remove();
+    this.postText.remove();
+  }
+
+  protected override async populateContainer() {
+    const phonescreenTexture = await Assets.load(
+      MINIGAME_ASSET_ALIASES.PHONE_SCREEN,
+    );
+    const bg = new Sprite(phonescreenTexture);
+    bg.setSize(this.app.screen);
+    this.container.addChild(bg);
+
+    const postImageTexture = await Assets.load(
+      [
+        MINIGAME_ASSET_ALIASES.INSTA_POST_1,
+        MINIGAME_ASSET_ALIASES.INSTA_POST_2,
+        MINIGAME_ASSET_ALIASES.INSTA_POST_3,
+      ][this.week],
+    );
+    const post = new Sprite(postImageTexture);
+    const aspectRatio = post.height / post.width;
+    const postWidth = this.app.screen.width * 0.22;
+    post.width = postWidth;
+    post.height = postWidth * aspectRatio;
+    post.anchor = 0.5;
+    post.x = this.app.screen.width / 2 + this.app.screen.width * 0.028;
+    post.y = this.app.screen.height * 0.38;
+    this.container.addChild(post);
+
+    this.remainingWords.clear();
+    for (const word of TARGET_SENTENCES[this.week].split(" ")) {
+      if (this.remainingWords.has(word)) continue;
+      this.remainingWords.add(word);
+      this.trie.insert(word);
+      this.addWordToScreen(word);
+      const displayWord = document.createElement("span");
+      displayWord.dataset.word = word;
+      displayWord.style.opacity = "0";
+      displayWord.innerText = word + " ";
+      this.postText.append(displayWord);
+    }
+
+    this.trie.insert("");
+    this.currentNode = this.trie.find("");
+  }
+
+  protected override onKeyDown(key: string) {
```

Comment:
I was wondering if we should skip repeat characters? Some people's keyboards might be sensitive and make it hard to beat the minigame